### ISSUE TITLE
After upgrading a 16.04, no storagerouters or vpools could be loaded.

### ISSUE THREAD

**Issue by kinvaris** [openvstorage/framework#1160]
After upgrading a fargo from previous monday (7/nov) to today (15/nov) I noticed that I could not load my storagerouters nor the vPools anymore.

After some digging I noticed these exceptions:

* Loading the vPools:
```
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]: , sdch"}, "request": {"timestamp": "1479217273325", "vpool_guid": "40fc85e5-b5e7-4d99-9abf-5a463f7bc1e4", "contents": "storagerouter,vpool_backend_info,vdisks_guids"}}
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]: 2016-11-15 14:41:13 46100 +0100 - stor-06.be-g8-4 - 28110/139966401392144 - api/middleware - 2740 - ERROR - An unhandled exception occurred: 'caching_info'
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]: Traceback (most recent call last):
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/django/core/handlers/base.py", line 132, in get_response
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     response = wrapped_callback(request, *callback_args, **callback_kwargs)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/viewsets.py", line 78, in view
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     return self.dispatch(request, *args, **kwargs)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     return view_func(*args, **kwargs)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 399, in dispatch
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     response = self.handle_exception(exc)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 396, in dispatch
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     response = handler(request, *args, **kwargs)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 519, in new_function
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     return_value = f(*args, **kwargs)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 84, in new_function
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     return f(*args, **kw)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 312, in new_function
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     data = FullSerializer(object_type, contents=contents, instance=data_list, many=True).data
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/serializers.py", line 571, in data
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     self._data = [self.to_native(item) for item in obj]
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/serializers.py", line 349, in to_native
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     value = field.field_to_native(obj, field_name)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/fields.py", line 198, in field_to_native
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     value = get_component(value, component)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/fields.py", line 56, in get_component
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     val = getattr(obj, attr_name)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 344, in <lambda>
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     fget = lambda s: s._get_dynamic_property(dynamic)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 413, in _get_dynamic_property
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     return self._backend_property(data_loader, dynamic)
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 953, in _backend_property
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     dynamic_data = function()
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/ovs/dal/hybrids/storagedriver.py", line 173, in _vpool_backend_info
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]:     if self.storagerouter_guid in self.vpool.metadata['backend']['caching_info']:
Nov 15 14:41:13 stor-06.be-g8-4 gunicorn[27590]: KeyError: 'caching_info'
```

* Loading the storagerouters:
```
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]: 2016-11-15 14:21:10 30400 +0100 - stor-06.be-g8-4 - 28056/139965982829456 - api/middleware - 391 - ERROR - An unhandled exception occurred: u'READ'
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]: Traceback (most recent call last):
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/django/core/handlers/base.py", line 132, in get_response
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     response = wrapped_callback(request, *callback_args, **callback_kwargs)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/viewsets.py", line 78, in view
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     return self.dispatch(request, *args, **kwargs)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     return view_func(*args, **kwargs)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 399, in dispatch
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     response = self.handle_exception(exc)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/views.py", line 396, in dispatch
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     response = handler(request, *args, **kwargs)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 519, in new_function
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     return_value = f(*args, **kwargs)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 84, in new_function
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     return f(*args, **kw)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/webapps/api/backend/decorators.py", line 363, in new_function
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     return Response(FullSerializer(object_type, contents=contents, instance=obj).data, status=return_status)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/serializers.py", line 573, in data
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     self._data = self.to_native(obj)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/serializers.py", line 349, in to_native
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     value = field.field_to_native(obj, field_name)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/fields.py", line 198, in field_to_native
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     value = get_component(value, component)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/usr/lib/python2.7/dist-packages/rest_framework/fields.py", line 56, in get_component
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     val = getattr(obj, attr_name)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 344, in <lambda>
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     fget = lambda s: s._get_dynamic_property(dynamic)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 413, in _get_dynamic_property
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     return self._backend_property(data_loader, dynamic)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 953, in _backend_property
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     dynamic_data = function()
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:   File "/opt/OpenvStorage/ovs/dal/hybrids/storagerouter.py", line 124, in _partition_config
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]:     dataset[role].append(partition.guid)
Nov 15 14:21:10 stor-06.be-g8-4 gunicorn[27590]: KeyError: u'READ'
```

After some discussion with @khenderick & @kvanhijf, it seemed like the migration code did not run properly. 

After @kvanhijf created a hotfix, and reran the migration it was fixed. 
#1159

**Comment by khenderick** [comment]
Fixed by #1159, packaged in openvstorage-2.7.5-fargo.2 and openvstorage-2.7.6-rev.4301.8e25118

**Comment by kinvaris** [comment]
PASSED: executed successfully on a gig environment
